pkg/rancher-desktop/backend/k3sHelper.ts

```diff
@@ -321,11 +323,17 @@ export default class K3sHelper extends events.EventEmitter {
     }
 
     // Check that this release has all the assets we expect.
-    if (Object.values(this.filenames).every(name => entry.assets.some(v => v.name === name))) {
-      console.debug(`Adding version ${ version.raw }`);
-      this.versions[version.version] = new VersionEntry(version);
-    } else {
-      console.debug(`Skipping version ${ version.raw } due to missing files`);
+    if (entry.assets.find(ea => ea.name === this.filenames.exe)) {
```

Comment:
I prefer avoiding unnecessarily nested `if`s:
```
diff --git a/pkg/rancher-desktop/backend/k3sHelper.ts b/pkg/rancher-desktop/backend/k3sHelper.ts
index a98b26af..fd8b40a1 100644
--- a/pkg/rancher-desktop/backend/k3sHelper.ts
+++ b/pkg/rancher-desktop/backend/k3sHelper.ts
@@ -323,16 +323,15 @@ export default class K3sHelper extends events.EventEmitter {
     }
 
     // Check that this release has all the assets we expect.
-    if (entry.assets.find(ea => ea.name === this.filenames.exe)) {
-      if (entry.assets.find(ea => ea.name === this.filenames.checksum)) {
-        const foundImage = this.filenames.images.find(name => entry.assets.some(v => v.name === name));
+    if (entry.assets.find(ea => ea.name === this.filenames.exe) &&
+        entry.assets.find(ea => ea.name === this.filenames.checksum)) {
+      const foundImage = this.filenames.images.find(name => entry.assets.some(v => v.name === name));
 
-        if (foundImage) {
-          this.versions[version.version] = new VersionEntry(version);
-          console.log(`Adding version ${ version.raw } - ${ foundImage }`);
-        } else {
-          console.debug(`Skipping version ${ version.raw } due to missing files`);
-        }
+      if (foundImage) {
+        this.versions[version.version] = new VersionEntry(version);
+        console.log(`Adding version ${ version.raw } - ${ foundImage }`);
+      } else {
+        console.debug(`Skipping version ${ version.raw } due to missing files`);
       }
     }
 
```

Reply:
changed.  For testing, Jan has written acceptance criteria in issue #1680.